scripts/wavie.js

```diff
@@ -0,0 +1,32 @@
+// Description
+//   Wave people as they enter and exit the room
+
+module.exports = function (robot) {
+    // Wave at the latest message in the room
+    function wave(room) {
+        // Make sure we have access to all the clients we need
+        if (!robot.adapter.client || !robot.adapter.client.web) {
+            return;
+        }
+
+        // Shorten web client so we can have nice concise lines :)
+        webClient = robot.adapter.client.web;
+
+        // Try channel as a public channel search first, then as a private channel
+        webClient.channels.info(room)
+            .then(res => webClient.reactions.add('wave', {channel: room, timestamp: res.channel.latest.ts}))
+            .catch(err => { if (err == 'channel_not_found') return webClient.groups.info(room); })
+            .then(res => webClient.reactions.add('wave', {channel: room, timestamp: res.channel.latest.ts}))
```

Comment:
By the time we've had a failure and re-requested, there's a reasonable chance there'll be an extra message in the channel. We should be able to get the original message timestamp from the argument to the `robot.enter`/`robot.leave` callback.

Reply:
yeah I had that code initially but decided it overcomplicated things for (what I thought) was pretty edge case, but I think you're right.

Unfortunately, the result passed to `robot.enter`/`robot.leave` doesn't actually include a TextMessage that I can dig into (see https://github.com/hubotio/hubot/blob/master/src/message.js#L55). I was going to get this information using the `latest` attribute returned from `channels.info` and `groups.info` to check if the sub type is one of a user entering/leaving before waving, but this would mean no :wave: would occur in an "extra message" situation. Another option is something like `channels.history`, but I believe this requires additional bot privileges

Reply:
Good enough